server/src/main/java/org/elasticsearch/cluster/metadata/DiffableStringMap.java

```diff
@@ -65,47 +65,59 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public Diff<DiffableStringMap> diff(DiffableStringMap previousState) {
-        return new DiffableStringMapDiff(previousState, this);
+        if (this.equals(previousState)) {
+            return DiffableStringMapDiff.EMPTY;
+        }
+        final List<String> tempDeletes = new ArrayList<>();
+        final Map<String, String> tempUpserts = new HashMap<>();
```

Comment:
I think it's better to declare `tempUpserts` before the 2nd loop where it's actually used.

Reply:
I know the code part was just moved, but I still think it makes sense to improve it a bit.